process/block/preprocess/selectionSession.go

```diff
@@ -38,67 +39,80 @@ func NewSelectionSession(args ArgsSelectionSession) (*selectionSession, error) {
 	return &selectionSession{
 		accountsAdapter:        args.AccountsAdapter,
 		transactionsProcessor:  args.TransactionsProcessor,
-		ephemeralAccountsCache: make(map[string]vmcommon.AccountHandler),
+		ephemeralAccountsCache: make(map[string]state.UserAccountHandler),
 	}, nil
 }
 
-// GetAccountState returns the state of an account.
-// Will be called by mempool during transaction selection.
-func (session *selectionSession) GetAccountState(address []byte) (state.UserAccountHandler, error) {
-	account, err := session.getExistingAccount(address)
+// GetAccountNonceAndBalance returns nonce and balance of an account.
+// Will be called by the transactions pool, during transactions selection.
+func (session *selectionSession) GetAccountNonceAndBalance(address []byte) (uint64, *big.Int, bool, error) {
+	account, err := session.getCachedUserAccount(address)
 	if err != nil {
-		// TODO: brainstorm whether we should move the logic around "ErrAccNotFound",
```

Comment:
We're propagating that information as the third output parameter of `GetAccountNonceAndBalance`.